packer-provisioner-goss.go

```diff
@@ -117,7 +121,15 @@ func (p *Provisioner) Prepare(raws ...interface{}) error {
 	}
 
 	if p.config.DownloadPath == "" {
-		p.config.DownloadPath = fmt.Sprintf("/tmp/goss-%s-linux-%s", p.config.Version, p.config.Arch)
+		if p.config.URL == "" {
+			p.config.DownloadPath = fmt.Sprintf("/tmp/goss-%s-linux-%s", p.config.Version, p.config.Arch)
+		} else {
+			list := strings.Split(p.config.URL, "/")
+			arch := strings.Split(list[len(list)-1], "-")[2]
+			version := string([]rune(list[len(list)-2])[1:])
```

Comment:
Do you think this would ever be anything other than a `v`?  I don't have a strong feeling (and this may be more correct), but it would be easier to read with `TrimPrefix`.

Reply:
I can live with this change.